whisper_ner/dataset.py

```diff
@@ -0,0 +1,307 @@
+import json
+import logging
+import random
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Union
+
+import torch
+import torchaudio
+from transformers import WhisperProcessor
+
+
+from whisper_ner.utils import set_logger, token_padding
+
+set_logger()
+SAMPLE_RATE = 16_000
+
+
+class WhisperNERDataset(torch.utils.data.Dataset):
+    def __init__(
+        self,
+        data_path,
+        processor: WhisperProcessor,
+        max_seq_length: int = 448,
+        target_sample_rate=16_000,
+        audio_root_dir=None,
+        entity_dropout_prob=0.1,
+        max_samples=None,
+        n_neg_samples=2,
+        ner_mask_pct=0.,
+    ):
+        # read json file + create dataset
+        self.data_path = data_path
+        with open(data_path, "r") as f:
+            dataset = json.load(f)
+
+        if max_samples is not None:
+            logging.info(f"Using only {max_samples} samples from the dataset.")
+            dataset = dataset[:max_samples]
+
+        self.dataset = dataset
+        all_ner_tags = [
+            tag for d in dataset
+            for _, _, tag, _, _ in [self._unify_ner_tag(t) for t in d["ner"]]
```

Comment:
looks a bit ugly instead of `_, _, tag, _, _` use single param and slice using [2] index.